gluten-core/src/main/scala/org/apache/gluten/extension/GlutenPlan.scala

```diff
@@ -65,7 +65,13 @@ trait GlutenPlan extends SparkPlan with Convention.KnownBatchType with LogLevelU
   final def doValidate(): ValidationResult = {
     try {
       TransformerState.enterValidation
-      val res = doValidateInternal()
+      val res = BackendsApiManager.getValidatorApiInstance
+        .doSchemaValidate(schema)
+        .map {
+          reason =>
+            ValidationResult.notOk(s"Found schema check failure for $schema, due to: $reason")
+        }
+        .getOrElse(doValidateInternal())
```

Comment:
It would be better to move schema check before `TransformerState.enterValidation`. The transformer state is required by `tranform` method called by `doValidateInternal`. So before `doValidateInternal` is called, we don't need to set that state.

Reply:
Updated, thanks.